docs/source/install-script.rst

```diff
@@ -0,0 +1,141 @@
+:orphan: true
+
+.. role:: bash(code)
+   :language: bash
+
+=================
+Firedrake Install
+=================
+
+The structure of the Firedrake install process consists of the following steps:
+
+1. Installing system dependencies
+2. Building and configuring PETSc
+3. Installing Python dependencies  
+
+These steps are only supported for for Linux and MacOS distributions. 
+Further we will take the above steps and discuss them in detail. 
+
+Installing system dependencies
+-------------------------------
+
+Linux
+~~~~~
+
+For Linux distributions the system dependencies are installed via the :bash:`apt-get` package manager. The necessary dependencies for running default firedrake are:
+
+- build-essential
+- autoconf
+- automake
+- bison
+- flex
+- cmake
+- gfortran
+- git
+- libtool
+- python3-dev
+- python3-pip
+- python3-tk
+- python3-venv
+- zlib1g-dev
+- libboost-dev
+- libopenblas-dev (if no configuration is provided by the user)
+
+As a small observation, packages like *bison* and *flex* are used for *PT-Scotch* (tool for efficient parallel graph ordering).
+
+MacOS
+~~~~~
+
+For MacOs distributions (based on Darwin) the install script uses the :bash:`homebrew` package manager to install the following necessary dependencies:
+
+- `gcc`
+- `autoconf`
+- `make`
+- `automake`
+- `cmake`
+- `libtool`
+- `boost`
+- `openblas` (if no configuration is provided by the user)
+
+Additionally the install script will also want to install xcode-command line tools.
+
+After the installation of the system requirements is done, the installation script will then create a virtual python environment and prepare itself to install all its software dependencies there.
+
+Building and configuring PETSc
+-------------------------------
+
+One important package on which Firedrake software is built is the `PETSc_ <https://petsc.org/release/>`_ library, which offers a toolkit when dealing with partial differential equations. 
+The install script enables some customization for installing the PETSc, but also offers the user the option to use a previous installation of the library via the parameter :bash:`hounour_petsc`.
+If no previous previous installation is provided, the installation script will then build the PETSc package from source in the *src*  directory within the virtual python environment.
+
+The basic options for installing the library using the install argument :bash:`--minimal_petsc` are:
```

Comment:
`--minimal-petsc`

Reply:
I think this is also what I wrote

Reply:
You used an underscore instead of a hyphen.

Reply:
ooo :)